src/khoj/search_type/text_search.py

```diff
@@ -174,6 +174,7 @@ def collate_results(hits, entries: List[Entry], count=5) -> List[SearchResponse]
 
 def setup(
     text_to_jsonl: Type[TextToJsonl],
+    files: dict[str, str],
```

Comment:
Note: Should we type the `files` argument into a class, at least once these changes have stabilized?

Reply:
Certainly, once the types have been stabilized, it makes sense to move them into a class with `pydantic`.